packages/fast-components-styles-msft/src/button/index.ts

```diff
@@ -1,16 +1,150 @@
 import { IDesignSystem } from "../design-system";
 import { ComponentStyles } from "@microsoft/fast-jss-manager";
 import { IButtonClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { toPx } from "../utilities/units";
+import { applyType } from "../utilities/typography";
+import * as Chroma from "chroma-js";
 
 const styles: ComponentStyles<IButtonClassNameContract, IDesignSystem> = {
     button: {
+        ...applyType("t7", "vp1"),
+        maxWidth: toPx(374),
+        minWidth: toPx(120),
+        display: "inline-block",
+        padding: `${toPx(13)} ${toPx(12)} ${toPx(12)}`,
+        border: `${toPx(2)} solid transparent`,
+        borderRadius: toPx(2),
+        cursor: "pointer",
+        overflow: "hidden",
+        lineHeight: "1",
+        textAlign: "center",
+        whiteSpace: "nowrap",
+        verticalAlign: "bottom",
+        transition: "all 0.2s ease-in-out",
         color: (config: IDesignSystem): string => {
-            return config.foregroundColor;
+            return config.backgroundColor;
+        },
+        backgroundColor: (config: IDesignSystem): string => {

```

Comment:
Is this adding `gray` as a global config? I think we want to manipulate black values to get the desired color rather than adding a global for a color like gray.

Reply:
Definitely open to suggestions - this was the first instance of something like this. We could probably use Chroma to do it.

Reply:
lets try and manipulate the existing values - can we extrapolate the repeated parts of this code out into a function?